ddtrace/contrib/aiohttp/patch.py

```diff
@@ -1,40 +1,375 @@
+import asyncio
+import functools
+import logging
+import sys
+
 from ddtrace import config
 from ddtrace.vendor import wrapt
 
-from ...pin import Pin
 from ...utils.wrappers import unwrap
+from ...propagation.http import HTTPPropagator
+from ...pin import Pin
+from ...ext import http as ext_http
+from ..httplib.patch import should_skip_request
 
+import aiohttp
+from yarl import URL
 
 try:
     # instrument external packages only if they're available
     import aiohttp_jinja2
     from .template import _trace_render_template
-
-    template_module = True
 except ImportError:
-    template_module = False
+    _trace_render_template = None
+
+
+log = logging.getLogger(__name__)
+PY_35 = sys.version_info >= (3, 5)
+
+
+# NOTE: this will create a trace for the outer request, and a span for each
+#       connect (redirect), and optionally a span for the read of the body
+
+def _get_url_obj(obj):
+    url_obj = obj.url
+
+    if not isinstance(url_obj, URL):
+        url_obj = getattr(obj, 'url_obj', None)  # 1.x
+
+    return url_obj
+
+
+def _get_service_fallback(pin):
```

Comment:
This shouldn't be necessary since this is what the tracer currently does with `tracer.trace()` and `tracer.start_span`

Reply:
I'm sure things have changed much in the years since this PR has been open :)